### ISSUE TITLE
Revise `Relation` interface

### ISSUE THREAD

**Issue by notimetoname** [nspcc-dev/neofs-sdk-go#416]
## TL;DR

[`Relations`](https://github.com/nspcc-dev/neofs-sdk-go/blob/9d25bc7519bce3366212a91ef0492812b0320662/object/relations/relations.go#L21) is too complex, it is hard to understand who can call it for [listings](https://github.com/nspcc-dev/neofs-sdk-go/blob/9d25bc7519bce3366212a91ef0492812b0320662/object/relations/relations.go#L49-L55).

## Context

There was an [issue](https://github.com/nspcc-dev/neofs-sdk-go/issues/360) at first. In short, it requested to fix some NeoFS things related to child objects the same way `neofs-node` does. Discussion led to a helper function that can take some necessary "network" interface, ask for some meta info and return `[]oid.ID` with child objects. Why? Cause more than one place requests for such functionality. https://github.com/nspcc-dev/neofs-sdk-go/pull/361 was born and merged. The pool was and is the only implementation of the interface.

## Problems

1. I did SDK update in the `neofs-node` and i did not want to implement `Relations`.
2. Looks like it is not a helper SDK func if it requests to implement not that simple methods.
3. Looks like it forced `Pool` to implement some unsolicited methods.
4. Well, looks like it requests more than it helps: it is just a few `for` loops inside if everything is already implemented.

## Suggestions

1. Make `ListRelations` more complex but require less by its args (make `Relations` more simple).
2. Make `Relations` that simple that `Pool` and `Client` implement it as is, without any additional methods (well, it is not a `Relations` interface then, must be renamed).
3. Drop any unrequested `Pool`s methods (such as `GetLeftSibling`).
4. Make all the changes in sync with https://github.com/nspcc-dev/neofs-sdk-go/issues/380.
